### ISSUE TITLE
5.20 blocker: inappropriate conversion of I32 to SSize_t

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#13580]
Migrated from [rt.perl.org#121182](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=121182) (status was 'resolved')

Searchable as RT121182$

**Comment by p5pRT** [comment]
### From @demerphq

Recently Brian Fraser has been compiling Perl with
\-Accflags="\-ftrapv"\, and has been pointing out overlow issues that it
reveals\.

Compiling with this option will not even build as it encounters an
error in regexec\.c line 746 during the build process\.

const SSize\_t eshift = CHR\_DIST\(\(U8\*\)strend\, \(U8\*\)s\) \- end;

This is reproducable with

\./miniperl \-Dr \-e'"path\_sep&#8203;:\\t8490"=~m/^\(\\S\+\)&#8203;:\\s\*\(\\d\+\)$/'

That SSize\_t used to be a I32 which was changed in&#8203;:
ea3daa5d5f60389b105c00d4f7d05b5c2f2155f2 by Father Chrysostomos\.

Other than that this code has been more or less unchanged for a looong time\.

I dont know if it is the change from SSize\_t to I32 or something else\,
but IMO this should be a blocker for perl 5\.20\.

Yves

\-\- 
perl \-Mre=debug \-e "/just|another|perl|hacker/"

**Comment by p5pRT** [comment]
### From @iabyn

On Wed\, Feb 05\, 2014 at 04&#8203;:04&#8203;:37AM \-0800\, yves orton wrote&#8203;:
> \# New Ticket Created by  yves orton 
> \# Please include the string&#8203;:  \[perl \#121182\]
> \# in the subject line of all future correspondence about this issue\. 
> \# \<URL&#8203;: https://rt-archive.perl.org/perl5/Ticket/Display.html?id=121182 >
> 
> 
> Recently Brian Fraser has been compiling Perl with
> \-Accflags="\-ftrapv"\, and has been pointing out overlow issues that it
> reveals\.
> 
> Compiling with this option will not even build as it encounters an
> error in regexec\.c line 746 during the build process\.
> 
> const SSize\_t eshift = CHR\_DIST\(\(U8\*\)strend\, \(U8\*\)s\) \- end;
> 
> This is reproducable with
> 
> \./miniperl \-Dr \-e'"path\_sep&#8203;:\\t8490"=~m/^\(\\S\+\)&#8203;:\\s\*\(\\d\+\)$/'
> 
> That SSize\_t used to be a I32 which was changed in&#8203;:
> ea3daa5d5f60389b105c00d4f7d05b5c2f2155f2 by Father Chrysostomos\.
> 
> Other than that this code has been more or less unchanged for a looong time\.
> 
> I dont know if it is the change from SSize\_t to I32 or something else\,
> but IMO this should be a blocker for perl 5\.20\.

This code has changed out of all recognition in my intuit branch\,
and so probably isn't an issue any more\. But I'll make a note to compile
with \-ftrapv before merging back\.

\-\- 
"Strange women lying in ponds distributing swords is no basis for a system
of government\. Supreme executive power derives from a mandate from the
masses\, not from some farcical aquatic ceremony\."
&nbsp;    \-\- Dennis\, "Monty Python and the Holy Grail"

**Comment by p5pRT** [comment]
The RT System itself - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
### From @demerphq

On 5 February 2014 20&#8203;:25\, Dave Mitchell via RT
\<perlbug\-followup@&#8203;perl\.org> wrote&#8203;:
> On Wed\, Feb 05\, 2014 at 04&#8203;:04&#8203;:37AM \-0800\, yves orton wrote&#8203;:
>> \# New Ticket Created by  yves orton
>> \# Please include the string&#8203;:  \[perl \#121182\]
>> \# in the subject line of all future correspondence about this issue\.
>> \# \<URL&#8203;: https://rt-archive.perl.org/perl5/Ticket/Display.html?id=121182 >
>>
>>
>> Recently Brian Fraser has been compiling Perl with
>> \-Accflags="\-ftrapv"\, and has been pointing out overlow issues that it
>> reveals\.
>>
>> Compiling with this option will not even build as it encounters an
>> error in regexec\.c line 746 during the build process\.
>>
>> const SSize\_t eshift = CHR\_DIST\(\(U8\*\)strend\, \(U8\*\)s\) \- end;
>>
>> This is reproducable with
>>
>> \./miniperl \-Dr \-e'"path\_sep&#8203;:\\t8490"=~m/^\(\\S\+\)&#8203;:\\s\*\(\\d\+\)$/'
>>
>> That SSize\_t used to be a I32 which was changed in&#8203;:
>> ea3daa5d5f60389b105c00d4f7d05b5c2f2155f2 by Father Chrysostomos\.
>>
>> Other than that this code has been more or less unchanged for a looong time\.
>>
>> I dont know if it is the change from SSize\_t to I32 or something else\,
>> but IMO this should be a blocker for perl 5\.20\.
>
> This code has changed out of all recognition in my intuit branch\,
> and so probably isn't an issue any more\. But I'll make a note to compile
> with \-ftrapv before merging back\.

So I should stop debugging then?

I understand the cause of the overflow now sortof&#8203;:
prog\->check\_offset\_max is set to SSize\_t\_MAX\, which is then added to
the length of the check string \(1\)\, which then overflows\. \(I got the
line that overflows wrong by one line\)\.

I played around with this and tweaked the pattern from ^\(\\S\+\) to
^\(\\S\{1\,9\}\) to see what happens\, and to me it looks totally wrong&#8203;:

$ \./miniperl \-Dr \-e'"path\_sep&#8203;:\\t8490"=~m/^\(\\S\{1\,9\}\)&#8203;:\\s\*\(\\d\+\)$/'
Compiling REx "^\(\\S\{1\,9\}\)&#8203;:\\s\*\(\\d\+\)$"
rarest char : at 0
synthetic stclass
"ANYOF\[\\x\{00\}\-\\x\{08\}\\x\{0E\}\-\\x\{1F\}\\x\{21\}\-\\x\{FF\}\]\[\{utf8\}0100\-167F
1681\-1FFF 200B\-2027 202A\-202E 2030\-205E 2060\-2FFF 3001\-INFINITY\]"\.
Final program&#8203;:
&nbsp;   1&#8203;: BOL \(2\)
&nbsp;   2&#8203;: OPEN1 \(4\)
&nbsp;   4&#8203;:   CURLY \{1\,9\} \(7\)
&nbsp;   6&#8203;:     NPOSIXD\[\\s\] \(0\)
&nbsp;   7&#8203;: CLOSE1 \(9\)
&nbsp;   9&#8203;: EXACT \<&#8203;:> \(11\)
&nbsp;  11&#8203;: STAR \(13\)
&nbsp;  12&#8203;:   POSIXD\[\\s\] \(0\)
&nbsp;  13&#8203;: OPEN2 \(15\)
&nbsp;  15&#8203;:   PLUS \(17\)
&nbsp;  16&#8203;:     POSIXD\[\\d\] \(0\)
&nbsp;  17&#8203;: CLOSE2 \(19\)
&nbsp;  19&#8203;: EOL \(20\)
&nbsp;  20&#8203;: END \(0\)
floating "&#8203;:" at 1\.\.9223372036854775807 \(checking floating\)

^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Why does it say 9223372036854775807 here? The check\_offset\_max should
be 9 or so\, not "infinity"\.

stclass ANYOF\[\\x\{00\}\-\\x\{08\}\\x\{0E\}\-\\x\{1F\}\\x\{21\}\-\\x\{FF\}\]\[\{utf8\}0100\-167F
1681\-1FFF 200B\-2027 202A\-202E 2030\-205E 2060\-2FFF 3001\-INFINITY\]
anchored\(BOL\) minlen 3
Enabling $\` $& $' support \(0x7\)\.

^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
This one liner does not use $\` not $& not $'\. Why are we enabling this support?

EXECUTING\.\.\.

Matching REx "^\(\\S\{1\,9\}\)&#8203;:\\s\*\(\\d\+\)$" against "path\_sep&#8203;:%t8490"
Aborted

IMO this should not overflow\. And it should not try to match '&#8203;:' at
infinity\. It should try it to match it at position 1\.\.9

BTW\, I mistitled this bug report\. I no longer think this is directly
related to the use of SSize\_t\.

Cheers\,
Yves

\-\- 
perl \-Mre=debug \-e "/just|another|perl|hacker/"

**Comment by p5pRT** [comment]
### From @demerphq

On 5 February 2014 20&#8203;:38\, demerphq \<demerphq@&#8203;gmail\.com> wrote&#8203;:
> On 5 February 2014 20&#8203;:25\, Dave Mitchell via RT
> \<perlbug\-followup@&#8203;perl\.org> wrote&#8203;:
>> On Wed\, Feb 05\, 2014 at 04&#8203;:04&#8203;:37AM \-0800\, yves orton wrote&#8203;:
\[snip\]
>> This code has changed out of all recognition in my intuit branch\,
>> and so probably isn't an issue any more\. But I'll make a note to compile
>> with \-ftrapv before merging back\.

Trying with your branch things look better\, so I will stop digging\,
mostly\. BTW\, I like the improvements to the diagnostics for the start
class optimizations\.

But I still see the 1\.\.9223372036854775807 in there\, which IMO is just
wrong\. Maybe it is not in scope of your patch sequence tho?

I also see support for $\` $& $'\.

And I wonder what that "rarest char : at 0" is for\. I bet its dead
code related to study\.

Yves

This is what i see with your branch&#8203;:

$ \./miniperl \-Dr \-e'"path\_sep&#8203;:\\t8490"=~m/^\(\\S\{1\,9\}\)&#8203;:\\s\*\(\\d\+\)$/'
Compiling REx "^\(\\S\{1\,9\}\)&#8203;:\\s\*\(\\d\+\)$"
rarest char : at 0
synthetic stclass "ANYOF\[\\x\{00\}\-\\x\{08\}\\x\{0E\}\-\\x\{1F\}\\x\{21\}\-\\x\{7F\}\]"\.
Final program&#8203;:
&nbsp;   1&#8203;: BOL \(2\)
&nbsp;   2&#8203;: OPEN1 \(4\)
&nbsp;   4&#8203;:   CURLY \{1\,9\} \(7\)
&nbsp;   6&#8203;:     NPOSIXD\[\\s\] \(0\)
&nbsp;   7&#8203;: CLOSE1 \(9\)
&nbsp;   9&#8203;: EXACT \<&#8203;:> \(11\)
&nbsp;  11&#8203;: STAR \(13\)
&nbsp;  12&#8203;:   POSIXD\[\\s\] \(0\)
&nbsp;  13&#8203;: OPEN2 \(15\)
&nbsp;  15&#8203;:   PLUS \(17\)
&nbsp;  16&#8203;:     POSIXD\[\\d\] \(0\)
&nbsp;  17&#8203;: CLOSE2 \(19\)
&nbsp;  19&#8203;: EOL \(20\)
&nbsp;  20&#8203;: END \(0\)
floating "&#8203;:" at 1\.\.9223372036854775807 \(checking floating\) stclass
ANYOF\[\\x\{00\}\-\\x\{08\}\\x\{0E\}\-\\x\{1F\}\\x\{21\}\-\\x\{7F\}\] anchored\(BOL\) minlen 3
Enabling $\` $& $' support \(0x7\)\.

EXECUTING\.\.\.

Matching REx "^\(\\S\{1\,9\}\)&#8203;:\\s\*\(\\d\+\)$" against "path\_sep&#8203;:%t8490"
Intuit&#8203;: trying to determine minimum start position\.\.\.
&nbsp;  Found floating substr "&#8203;:" at offset 8\.\.\.
&nbsp;  try at start\.\.\.
&nbsp;  looking for class&#8203;: start\_shift&#8203;: 1 check\_at&#8203;: 8 s&#8203;: 0 endpos&#8203;: 8 checked\_upto&#8203;: 0
&nbsp;  Does not contradict STCLASS\.\.\.
Intuit&#8203;: Successfully guessed&#8203;: match at offset 0
&nbsp;   0 \<> \<path\_sep&#8203;:>          |  1&#8203;:BOL\(2\)
&nbsp;   0 \<> \<path\_sep&#8203;:>          |  2&#8203;:OPEN1\(4\)
&nbsp;   0 \<> \<path\_sep&#8203;:>          |  4&#8203;:CURLY \{1\,9\}\(7\)
&nbsp;                                  NPOSIXD\[\\s\] can match 9 times out of 9\.\.\.
&nbsp;   8 \<th\_sep> \<&#8203;:%t8490>      |  7&#8203;:  CLOSE1\(9\)
&nbsp;   8 \<th\_sep> \<&#8203;:%t8490>      |  9&#8203;:  EXACT \<&#8203;:>\(11\)
&nbsp;   9 \<th\_sep&#8203;:> \<%t8490>      | 11&#8203;:  STAR\(13\)
&nbsp;                                    POSIXD\[\\s\] can match 1 times out
of 2147483647\.\.\.
&nbsp;  10 \<th\_sep&#8203;:%t> \<8490>      | 13&#8203;:    OPEN2\(15\)
&nbsp;  10 \<th\_sep&#8203;:%t> \<8490>      | 15&#8203;:    PLUS\(17\)
&nbsp;                                      POSIXD\[\\d\] can match 4 times out
of 2147483647\.\.\.
&nbsp;  14 \<th\_sep&#8203;:%t8490> \<>      | 17&#8203;:      CLOSE2\(19\)
&nbsp;  14 \<th\_sep&#8203;:%t8490> \<>      | 19&#8203;:      EOL\(20\)
&nbsp;  14 \<th\_sep&#8203;:%t8490> \<>      | 20&#8203;:      END\(0\)
Match successful\!
Freeing REx&#8203;: "^\(\\S\{1\,9\}\)&#8203;:\\s\*\(\\d\+\)$"

**Comment by p5pRT** [comment]
### From @demerphq

On 5 February 2014 20&#8203;:58\, demerphq \<demerphq@&#8203;gmail\.com> wrote&#8203;:
> On 5 February 2014 20&#8203;:38\, demerphq \<demerphq@&#8203;gmail\.com> wrote&#8203;:
>> On 5 February 2014 20&#8203;:25\, Dave Mitchell via RT
>> \<perlbug\-followup@&#8203;perl\.org> wrote&#8203;:
>>> On Wed\, Feb 05\, 2014 at 04&#8203;:04&#8203;:37AM \-0800\, yves orton wrote&#8203;:
> \[snip\]
>>> This code has changed out of all recognition in my intuit branch\,
>>> and so probably isn't an issue any more\. But I'll make a note to compile
>>> with \-ftrapv before merging back\.
>
> Trying with your branch things look better\, so I will stop digging\,
> mostly\. BTW\, I like the improvements to the diagnostics for the start
> class optimizations\.
>
> But I still see the 1\.\.9223372036854775807 in there\, which IMO is just
> wrong\. Maybe it is not in scope of your patch sequence tho?

This changed sometime in 5\.9\.x\, in the latest 5\.8 it shows "1\.\.9"
which is what I expect\. In 5\.10\.0 it shows "1\.\.2147483647"\.

I wonder if I should open a new ticket for this\.\.\.

Anyway\, I am going to see if I can get bisect working to track down
which patch changed this\.

Cheers\,
Yves

**Comment by p5pRT** [comment]
### From @iabyn

On Wed\, Feb 05\, 2014 at 08&#8203;:58&#8203;:11PM \+0800\, demerphq wrote&#8203;:
> On 5 February 2014 20&#8203;:38\, demerphq \<demerphq@&#8203;gmail\.com> wrote&#8203;:
> Trying with your branch things look better\, so I will stop digging\,
> mostly\. BTW\, I like the improvements to the diagnostics for the start
> class optimizations\.

Note also that I've just pushed the davem/intuit2 branch\, which contains
my newer changes\.

> But I still see the 1\.\.9223372036854775807 in there\, which IMO is just
> wrong\. Maybe it is not in scope of your patch sequence tho?

Nope\, that a compile\-time thing\. I'm a runtime\-kind of guy; I'm just
concerned with re\_intuit\_start\(\) :\-\)\.

> I also see support for $\` $& $'\.

That's a "feature" under COW\. It unconditionally enables $\` $& $' support\,
since that no longer incurs a penalty\.

> And I wonder what that "rarest char : at 0" is for\. I bet its dead
> code related to study\.

It's produced by Perl\_fbm\_compile\(\)\, so I think its still live code\.

\[ Just as an aside\, why was the study / scream stuff removed? I'm not
doubting that it was sensible to do so\, it's just I don't know enough
about it to understand why \]


\-\- 
In economics\, the exam questions are the same every year\.
They just change the answers\.

**Comment by p5pRT** [comment]
### From @demerphq

On 5 February 2014 22&#8203;:08\, Dave Mitchell \<davem@&#8203;iabyn\.com> wrote&#8203;:
> On Wed\, Feb 05\, 2014 at 08&#8203;:58&#8203;:11PM \+0800\, demerphq wrote&#8203;:
>> On 5 February 2014 20&#8203;:38\, demerphq \<demerphq@&#8203;gmail\.com> wrote&#8203;:
>> Trying with your branch things look better\, so I will stop digging\,
>> mostly\. BTW\, I like the improvements to the diagnostics for the start
>> class optimizations\.
>
> Note also that I've just pushed the davem/intuit2 branch\, which contains
> my newer changes\.
>
>> But I still see the 1\.\.9223372036854775807 in there\, which IMO is just
>> wrong\. Maybe it is not in scope of your patch sequence tho?
>
> Nope\, that a compile\-time thing\. I'm a runtime\-kind of guy; I'm just
> concerned with re\_intuit\_start\(\) :\-\)\.

Heh\. Ok\. I probably broke it anyway\. :\-\)

>> I also see support for $\` $& $'\.
>
> That's a "feature" under COW\. It unconditionally enables $\` $& $' support\,
> since that no longer incurs a penalty\.

Ah\. Ok\.

>> And I wonder what that "rarest char : at 0" is for\. I bet its dead
>> code related to study\.
>
> It's produced by Perl\_fbm\_compile\(\)\, so I think its still live code\.

Its odd that we would call fbm\_compile on a 1 byte string\. There is no
advantage in using FBM matching on a single character\.

> \[ Just as an aside\, why was the study / scream stuff removed? I'm not
> doubting that it was sensible to do so\, it's just I don't know enough
> about it to understand why \]

I think there were a bunch of reasons&#8203;:

\* It didn't work on utf8\.
\* It slowed down matching short strings\, and only sped up some long
string matches\.
\* It didn't work in a bunch of other situations\.
\* It was hard to understand\.
\* It wasn't well tested\.

If you check the git log \-\-grep study then you will see many patches
dealing with study/scream issues that were not noticed because it was
used so infrequently\.

Basically we turned it off and nobody noticed\.\.\. Which says it wasn't
doing much :\-\)

Yves

\-\- 
perl \-Mre=debug \-e "/just|another|perl|hacker/"

**Comment by p5pRT** [comment]
### From @iabyn

On Wed\, Feb 05\, 2014 at 10&#8203;:35&#8203;:35PM \+0800\, demerphq wrote&#8203;:
> Its odd that we would call fbm\_compile on a 1 byte string\. There is no
> advantage in using FBM matching on a single character\.

No\, but it simplifies the runtime\. We always just call fbm\_instr\(\)
\(which has special handling for 0\,1 and 2\-length substrs\)\, rather than
having to worry about whether to call fbm\_instr\(\) each time\.

Although I'm not clear whether fbm\_compile\(\) handles this cleverly\, i\.e
in the case of 1\-element substrs if it skips adding the fbm magic?

I'll add it to my list of things to look at\.

> Basically we turned it off and nobody noticed\.\.\. Which says it wasn't
> doing much :\-\)

Ah\, ok thanks :\-\)

\-\- 
You never really learn to swear until you learn to drive\.

**Comment by p5pRT** [comment]
### From @demerphq

On 5 February 2014 22&#8203;:35\, demerphq \<demerphq@&#8203;gmail\.com> wrote&#8203;:
> On 5 February 2014 22&#8203;:08\, Dave Mitchell \<davem@&#8203;iabyn\.com> wrote&#8203;:
>> On Wed\, Feb 05\, 2014 at 08&#8203;:58&#8203;:11PM \+0800\, demerphq wrote&#8203;:
>>> On 5 February 2014 20&#8203;:38\, demerphq \<demerphq@&#8203;gmail\.com> wrote&#8203;:
>>> Trying with your branch things look better\, so I will stop digging\,
>>> mostly\. BTW\, I like the improvements to the diagnostics for the start
>>> class optimizations\.
>>
>> Note also that I've just pushed the davem/intuit2 branch\, which contains
>> my newer changes\.
>>
>>> But I still see the 1\.\.9223372036854775807 in there\, which IMO is just
>>> wrong\. Maybe it is not in scope of your patch sequence tho?
>>
>> Nope\, that a compile\-time thing\. I'm a runtime\-kind of guy; I'm just
>> concerned with re\_intuit\_start\(\) :\-\)\.
>
> Heh\. Ok\. I probably broke it anyway\. :\-\)

I did\.

304ee84bde82d4eee33b0d0ff03080b360eae72b is the first bad commit
commit 304ee84bde82d4eee33b0d0ff03080b360eae72b
Author&#8203;: Yves Orton \<demerphq@&#8203;gmail\.com>
Date&#8203;:   Thu Nov 30 02&#8203;:12&#8203;:25 2006 \+0100

&nbsp;    Re&#8203;: \[perl \#41010\] \(?\(COND\)\) in pattern matching not working properly
&nbsp;    Message\-ID&#8203;: \<9b18b3110611291612i137803f4lab2739e3d9af40f7@&#8203;mail\.gmail\.com>

&nbsp;    p4raw\-id&#8203;: //depot/perl@&#8203;29441

:100644 100644 adcdb84bfd8a08fab367cd0bd5410a099cecb520
1b9b5a4317e26bf496c0fbe05314aa929cb86226 M embed\.fnc
:100644 100644 8cc8bba9a300e692244b4bb848a3ef9c602efe82
5c6930fe79adfe00dbdf9cdf08017333506c9bd2 M embed\.h
:100644 100644 06bef08cea74cf9f7194caf6c11fbf58ce2657ec
4f8edf42ef2e17ae89771042531c9353cc4cc84b M proto\.h
:100644 100644 c5a0b7bef0746b57554963dcd5a767c5093c2275
359e4f6d2c30ad84e35f6712a8daff90d1c1750b M regcomp\.c
:040000 040000 19b0fc7a811878a1959a7128ad978a17c74e02a7
203e6ca992873d3c3655b16134c3f62a7e89d473 M t


\-\- 
perl \-Mre=debug \-e "/just|another|perl|hacker/"

**Comment by p5pRT** [comment]
### From @demerphq

On 5 February 2014 23&#8203;:41\, demerphq \<demerphq@&#8203;gmail\.com> wrote&#8203;:
> On 5 February 2014 22&#8203;:35\, demerphq \<demerphq@&#8203;gmail\.com> wrote&#8203;:
>> On 5 February 2014 22&#8203;:08\, Dave Mitchell \<davem@&#8203;iabyn\.com> wrote&#8203;:
>>> On Wed\, Feb 05\, 2014 at 08&#8203;:58&#8203;:11PM \+0800\, demerphq wrote&#8203;:
>>>> On 5 February 2014 20&#8203;:38\, demerphq \<demerphq@&#8203;gmail\.com> wrote&#8203;:
>>>> Trying with your branch things look better\, so I will stop digging\,
>>>> mostly\. BTW\, I like the improvements to the diagnostics for the start
>>>> class optimizations\.
>>>
>>> Note also that I've just pushed the davem/intuit2 branch\, which contains
>>> my newer changes\.
>>>
>>>> But I still see the 1\.\.9223372036854775807 in there\, which IMO is just
>>>> wrong\. Maybe it is not in scope of your patch sequence tho?
>>>
>>> Nope\, that a compile\-time thing\. I'm a runtime\-kind of guy; I'm just
>>> concerned with re\_intuit\_start\(\) :\-\)\.
>>
>> Heh\. Ok\. I probably broke it anyway\. :\-\)
>
> I did\.
>
> 304ee84bde82d4eee33b0d0ff03080b360eae72b is the first bad commit
> commit 304ee84bde82d4eee33b0d0ff03080b360eae72b
> Author&#8203;: Yves Orton \<demerphq@&#8203;gmail\.com>
> Date&#8203;:   Thu Nov 30 02&#8203;:12&#8203;:25 2006 \+0100
>
>     Re&#8203;: \[perl \#41010\] \(?\(COND\)\) in pattern matching not working properly
>     Message\-ID&#8203;: \<9b18b3110611291612i137803f4lab2739e3d9af40f7@&#8203;mail\.gmail\.com>
>
>     p4raw\-id&#8203;: //depot/perl@&#8203;29441
>
> :100644 100644 adcdb84bfd8a08fab367cd0bd5410a099cecb520
> 1b9b5a4317e26bf496c0fbe05314aa929cb86226 M embed\.fnc
> :100644 100644 8cc8bba9a300e692244b4bb848a3ef9c602efe82
> 5c6930fe79adfe00dbdf9cdf08017333506c9bd2 M embed\.h
> :100644 100644 06bef08cea74cf9f7194caf6c11fbf58ce2657ec
> 4f8edf42ef2e17ae89771042531c9353cc4cc84b M proto\.h
> :100644 100644 c5a0b7bef0746b57554963dcd5a767c5093c2275
> 359e4f6d2c30ad84e35f6712a8daff90d1c1750b M regcomp\.c
> :040000 040000 19b0fc7a811878a1959a7128ad978a17c74e02a7
> 203e6ca992873d3c3655b16134c3f62a7e89d473 M t

Fixed in&#8203;:

commit 66175af586bd8040b474a5d2ddb6e6308424b668
Author&#8203;: Yves Orton \<yves\.orton@&#8203;booking\.com>
Date&#8203;:   Thu Feb 6 01&#8203;:39&#8203;:30 2014 \+0800

&nbsp;    Fix regression in floating mandatory string optimisation

&nbsp;    In 304ee84bde82d4eee33b0d0ff03080b360eae72b I introduced a regression
&nbsp;    where floating mandatory strings which were at known offsets started
&nbsp;    being treated as though they were allowed "anywhere"\.

&nbsp;    This patch fixes the bug\, and adds tests to make sure it does not come
&nbsp;    back\.

&nbsp;    See also discussion in Perl RT \#121182\.

&nbsp;    https://rt.perl.org/Public/Bug/Display.html?id=121182


\-\- 
perl \-Mre=debug \-e "/just|another|perl|hacker/"

**Comment by p5pRT** [comment]
### From @tonycoz

On Wed Feb 05 09&#8203;:41&#8203;:50 2014\, demerphq wrote&#8203;:
> Fixed in&#8203;:
> 
> commit 66175af586bd8040b474a5d2ddb6e6308424b668
> Author&#8203;: Yves Orton \<yves\.orton@&#8203;booking\.com>
> Date&#8203;:   Thu Feb 6 01&#8203;:39&#8203;:30 2014 \+0800
> 
> Fix regression in floating mandatory string optimisation
> 
\.\.\.

Does that mean the ticket can be closed?

Should the subject change?

Tony

**Comment by p5pRT** [comment]
### From @demerphq

On 6 February 2014 06&#8203;:35\, Tony Cook via RT \<perlbug\-followup@&#8203;perl\.org> wrote&#8203;:
> On Wed Feb 05 09&#8203;:41&#8203;:50 2014\, demerphq wrote&#8203;:
>> Fixed in&#8203;:
>>
>> commit 66175af586bd8040b474a5d2ddb6e6308424b668
>> Author&#8203;: Yves Orton \<yves\.orton@&#8203;booking\.com>
>> Date&#8203;:   Thu Feb 6 01&#8203;:39&#8203;:30 2014 \+0800
>>
>> Fix regression in floating mandatory string optimisation
>>
> \.\.\.
>
> Does that mean the ticket can be closed?

Can you close this ticket and I will open two new ones\, one for the
regression I fixed\, and one for the still open issue that is resolved
by Dave's intuit branch\. Basically I consider that a blocker\, and I
consider it necessary for Dave's branch to be applied\.

Cheers\,
Yves

\-\- 
perl \-Mre=debug \-e "/just|another|perl|hacker/"

**Comment by p5pRT** [comment]
### From @tonycoz

On Wed Feb 05 18&#8203;:18&#8203;:40 2014\, demerphq wrote&#8203;:
> On 6 February 2014 06&#8203;:35\, Tony Cook via RT \<perlbug\-followup@&#8203;perl\.org>
> wrote&#8203;:
> > On Wed Feb 05 09&#8203;:41&#8203;:50 2014\, demerphq wrote&#8203;:
> >> Fixed in&#8203;:
> >>
> >> commit 66175af586bd8040b474a5d2ddb6e6308424b668
> >> Author&#8203;: Yves Orton \<yves\.orton@&#8203;booking\.com>
> >> Date&#8203;:   Thu Feb 6 01&#8203;:39&#8203;:30 2014 \+0800
> >>
> >> Fix regression in floating mandatory string optimisation
> >>
> > \.\.\.
> >
> > Does that mean the ticket can be closed?
> 
> Can you close this ticket and I will open two new ones\, one for the
> regression I fixed\, and one for the still open issue that is resolved
> by Dave's intuit branch\. Basically I consider that a blocker\, and I
> consider it necessary for Dave's branch to be applied\.

Resolved\.

Tony

**Comment by p5pRT** [comment]
@tonycoz - Status changed from 'open' to 'resolved'
